install.sh

```diff
@@ -33,123 +127,181 @@ valid_version() {
+  local compose_version_extracted
+  local compose_version_major
+  local compose_version_minor
+
+  compose_version_extracted=$(docker-compose --version | sed 's/^.*version\ //g' | sed 's/[\,\ ].*$//g' | sed 's/v//g')
+  compose_version_major=$(echo "${compose_version_extracted}" | awk -F. '{ print $1 }')
+  compose_version_minor=$(echo "${compose_version_extracted}" | awk -F. '{ print $2 }')
+
+  if ! _valid_version ${_COMPOSE_MINIMUM_VERSION_MAJOR} ${_COMPOSE_MINIMUM_VERSION_MINOR} "${compose_version_major}" "${compose_version_minor}"; then
+    _err "Minimum docker-compose version is ${_COMPOSE_MINIMUM_VERSION_MAJOR}.${_COMPOSE_MINIMUM_VERSION_MINOR}"
+    exit 1
+  fi
 }
 
-migrate_postgres_version_var() {
-  sed -i 's/\bPOSTGRES_PASSWORD/POSTGRES_ADMIN_PASSWORD/' customer.env
+_migrate_postgres_version_var() {
+  sed -i -e 's/POSTGRES_PASSWORD/POSTGRES_ADMIN_PASSWORD/g' customer.env
 }
 
-create_env_file() {
-    migrate_postgres_version_var
-    echo "[info] creating .env file..."
-    version=$(cat VERSION)
-    cat customer.env > .env
-    echo "" >> .env
-    echo "CARTO_SELFHOSTED_VERSION=$version" >> .env
-    cat env.tpl >> .env
-    mkdir -p certs
-    cp key.json certs/key.json
-    echo "[info] file .env successfully created"
-    echo "[info] script finished, run docker-compose up -d"
+_create_env_file() {
+  _migrate_postgres_version_var
+  _info "Creating .env file..."
+  local version
+  version=$(cat VERSION)
+  {
+    cat customer.env
+    echo ""
+    echo "CARTO_SELFHOSTED_VERSION=${version}"
+    cat env.tpl
+  } >.env
+  mkdir -p certs
+  cp key.json certs/key.json
+  _info "File .env successfully created"
+  _info "Script finished, run docker-compose up -d"
 }
 
-set -e
-
-# evaluate arguments
-if [ $# -gt 1 ]
-then
-  echo "[error] too many arguments, only one argument permitted"
-  echo "[help] usage: sh install.sh [--ignore-checks][--help]"
-  exit 1
-elif [ $# -eq 1 ] && [ "$1" = '--help' ]
-then
-  echo "[help] usage: sh install.sh [--ignore-checks][--help]"
-  exit 0
-elif [ $# -eq 1 ] && [ "$1" = '--ignore-checks' ]
-then
-  IGNORE_CHECKS=true
-elif [ $# -eq 1 ]
-then
-  echo "[error] invalid argument"
-  echo "[help] usage: sh install.sh [--ignore-checks][--help]"
-  exit 1
-else
-  IGNORE_CHECKS=false
-fi
-
-if [ $IGNORE_CHECKS = true ]
-then
-  echo "[info] skipping command checks..."
-  create_env_file
-  exit 0
-fi
-
-echo "[info] running command checks..."
-if ! command -v docker > /dev/null 2>&1
-then
-    echo "[error] docker is not installed, you can use the ./scripts/install_docker.sh helper"
-    exit 1
-fi
+_check_min_cloud_version() {
+  local MINIMAL_VERSION
+  local PACKAGE_VERSION
+  MINIMAL_VERSION=$1
+  PACKAGE_VERSION=$2
 
-check_docker_version
+  if [[ "$(echo "$MINIMAL_VERSION $PACKAGE_VERSION" | tr ' ' '\n' | sort -V | head -n1)" == $MINIMAL_VERSION ]]; then
+    cat <<-EOF
+      [error] minimum cloud version version is $MINIMAL_VERSION but your package was generated with $MINIMAL_VERSION
```

Comment:
Last `$MINIMAL_VERSION` shouldn't it be `$PACKAGE_VERSION `?

Reply:
yes! changed